### PR TITLE
Bugfix/too many form link results are ignored

### PR DESCRIPTION
**Pull request by Tom-Ritense**
Introduced single function to ensure retrieval is always returning 1 and only 1 otherwise an error state is masked.

Solution distilled from https://stackoverflow.com/questions/22694884/filter-java-stream-to-1-and-only-1-element

### CODE REVIEW

File: form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java [hunk 1]
```diff
@@ -453,4 +458,17 @@ private ObjectNode buildMetaDataObject(Document document) {
         metaDataNode.put("sequence", document.sequence());
         return metaDataNode;
     }
+
+    private static <T> Collector<T, ?, Optional<T>> single() {
```
Comment (marijnritense) [inline form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java:462]: Not entirely happy with this name, as it doesn't fully convey what this method is supposed to return. It returns a collector that results in one result if the list only has one entry. That's not clear when reading the method name.

Comment (Tom-Ritense) [reply form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java:462]: Do you have a suggestion for improving the name? from the stackoverflow I see a couple:

- onlyElement see [
](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/MoreCollectors.html#onlyElement--)- singleElementCollector
- zeroOrOne
- onlyOne
- toSingleton

Comment (marijnritense) [reply form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java:462]: singleElementCollector sounds best out of these options. I was playing with names like singleElementIfOnlyOneElementPresentCollector, but that becomes too cumbersome in my opinion.

File: form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java [hunk 2]
```diff
@@ -453,4 +458,17 @@ private ObjectNode buildMetaDataObject(Document document) {
...
+        return Collectors.collectingAndThen(
+            Collectors.toList(),
+            list -> {
+                if (list.size() == 1) {
+                    return Optional.of(list.get(0));
+                }
+                throw new IllegalStateException("Expected single result found: " + list.size());
```
[the remark this answers is not in the document]
Comment (Tom-Ritense) [reply form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java:469]: Ok will change.

File: form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java (hunk 1 above)
Comment (Tom-Ritense) [reply form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java:462]: singleElementCollector is fine then.

**Review by marijnritense — APPROVED** [review]

**Pull request merged**